src/main/java/com/project/appointments/service/ScheduleService.java

```diff
@@ -0,0 +1,69 @@
+package com.project.appointments.service;
+
+import com.project.appointments.model.entity.Schedule;
+import com.project.appointments.repository.ScheduleRepository;
+import jakarta.persistence.EntityNotFoundException;
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ScheduleService {
+  private final ScheduleRepository scheduleRepository;
+  private final AppointmentService appointmentService;
+
+  public List<Schedule> getSchedule() {
+    return scheduleRepository.findAll();
+  }
+
+  public Schedule getSchedule(Long id) {
+    return findScheduleById(id);
+  }
+
+  public Schedule addSchedule(Schedule schedule) {
+    return scheduleRepository.save(schedule);
+  }
+
+  public Schedule updateSchedule(Schedule schedule, Long id) {
+    Schedule saveSchedule = findScheduleById(id);
+    return scheduleRepository.save(saveSchedule.setDate(schedule.getDate()).setDoctorId(schedule.getDoctorId()));
+  }
+
+  public Schedule deleteSchedule(Long id) {
+    Schedule schedule = findScheduleById(id);
+    scheduleRepository.deleteById(id);
+    return schedule;
+  }
+  public void generateScheduleForThreeDays(Long doctorId){
+    LocalDate date = LocalDate.now();
+    scheduleSaveCycle(date,doctorId);
+  }
+  public void generateScheduleForThreeDaysTest(Long doctorId,LocalDate date){
+    scheduleSaveCycle(date,doctorId);
+  }
+  private void scheduleSaveCycle(LocalDate date,Long doctorId){
+    for(int i = 0; i<3;i++){
```

Comment:
According to the requirements "The schedule will be rendered every day at 8 am" 
If you run this function every day, duplicates will be generated. Think about how to rewrite and avoid it